# Hildebrand Group

## About
The Hildebrand Lab is a research group based at the Quadram Institute and
Earlham Institute in Norwich, UK.

We use metagenomics and develop associated bioinformatics algorithms to explore 
ecosystems from a global scale to the individual human gut microbiome.

For more detail see [falk.science](https://falk.science).

## Pipelines
We have developed a number of pipelines combining multiple pieces of software.
These are kept by member of the group on their personal accounts.
Below are links to pipelines developed in the group.

* **LotuS3**
([github](https://github.com/hildebra/lotus3)) 
16S/18S/ITS & other target amplicon sequencing pipeline for long & short read sequencing.

* **MG-TK**
([github](https://github.com/hildebra/MG-TK)) 
MetaGenomic ToolKit, pipeline handling end-to-end processing of metagenomes.

* **MG-STK**
([github](https://github.com/ksidorczuk/mg-stk))
(*repo currently private, request access from Katarzyna.Sidorczuk@quadram.ac.uk*)
MetaGenomic Strain ToolKit, tools which integrate with MG-TK to perform strain and strain enrichment analysis.

* **BM-TK**
([github](https://github.com/apduncan/bm-tk)) 
Batch prediction of base modfication for PacBio data from kinetics information.

* **gRAIL-TK**
([github](https://github.com/nabina99/gRAIL-TK))
(*repo currently private, request access from Nabina.Shrestha@quadram.ac.uk*)
Integrates guide RNA design, off target prediction, and target prioritization into an automated, configurable pipeline for CRISPR based genomic enrichment.

## Software
Software developed by members of the group are kept on their personal
github accounts.
Below are links to software developed in the group.

* **adhesiomeR**
([github](https://github.com/ksidorczuk/adhesiomeR)) 
([web](https://adhesiomer.quadram.ac.uk/))
Detect adhesion genes in *E. coli*, also available as a webserver.

* **cvanmf**
([github](https://github.com/apduncan/cvanmf)) 
([conda](https://anaconda.org/bioconda/cvanmf))
([docs](https://cvanmf.readthedocs.io)) 
Bicrossvalidated NMF clustering of data sets, with nextflow pipeline for
application to large datasets.

* **enterosignatures**
([web](https://enterosignatures.quadram.ac.uk/))
Webserver to quantify enterosginatures in gut microbiomes.

* **protal**
([github](https://github.com/4less/protal)) 
Alignment-based based strain-level metagenomic profiler.

* **benchpro**
([github](https://github.com/4less/benchpro)) 
Evaluation suite for taxonomic profiler performance.

* **varkit**
([github](https://github.com/4less/varkit)) 
k-mer-based strain-level metagenomic profiler.

* **DDT**
([github](https://github.com/hildebra/ddt))
(*repo currently private, request access from Falk.Hildebrand@quadram.ac.uk*)
Comprehensive numerical ecology statistics & analysis in R. 

* **RTK2**
([github](https://github.com/hildebra/rtk2)) 
([conda](https://anaconda.org/bioconda/rtk2))
Matrix operations on large metagenomic gene catalogues.

* **LCA**
([github](https://github.com/hildebra/LCA)) 
([conda](https://anaconda.org/bioconda/lca))
Least common ancestor taxonomy calculations.

* **SDM**
([github](https://github.com/hildebra/SDM)) 
([conda](https://anaconda.org/bioconda/sdm))
Demultiplexing and read quality filtering software.

* **MSAFix**
([github](https://github.com/hildebra/MSAfix)) 
Fixing frameshift errors in MSAs.

* **canopy2**
([github](https://github.com/hildebra/canopy2)) 
([conda](https://anaconda.org/bioconda/canopy))
Binning of metagenomic gene catalogues.

* **clusterMAGs**
([github](https://github.com/hildebra/clusterMAGs)) 
Binning of metagenomic gene catalogues.

* **rdCover**
([github](https://github.com/hildebra/rdCover)) 
Read coverage of genes, genomes etc. 


## Databases
* **KSGP**
([github](https://ksgp.earlham.ac.uk/)) 
16S Database of Archaea with deeper taxonomic resolution and GTDB format.

## Previous sofware versions
Some software or pipelines have been updated, below are links to previous
versions.

* **RTK**
([github](https://github.com/hildebra/Rarefaction)) 
([conda](https://anaconda.org/bioconda/rtk))
([cran](https://cran.r-project.org/web/packages/rtk/index.html))
Rarefaction and diversity analysis. Updated in RTK2.

* **LotuS2**
([github](https://github.com/hildebra/lotus2)) 
([web](http://lotus2.earlham.ac.uk))
([conda](http://lotus2.earlham.ac.uk))
16S/23S/ITS amplicon sequencing pipeline for illumina and PacBio sequences, supersed by LotuS3.

* **LotuS**
([web](http://psbweb05.psb.ugent.be/lotus)) 
Amplicon sequencing pipelines. Superseded by LotuS2.

* **MATAFILER**
([github](https://github.com/hildebra/MATAFILER)) 
Metagenomics pipeline, replaced by MG-TK.

